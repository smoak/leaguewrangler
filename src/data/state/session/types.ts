export interface Session {
  readonly userToken: string;
}
