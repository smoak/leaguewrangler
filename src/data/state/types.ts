import { Session } from './session/types';

export * from './session/types';

export interface ApplicationState {
  readonly session: Session;
}
