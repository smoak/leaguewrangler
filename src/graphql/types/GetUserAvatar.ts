/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: GetUserAvatar
// ====================================================

export interface GetUserAvatar_currentUser {
  __typename: "User";
  profilePhotoThumbnailUrl: string;
}

export interface GetUserAvatar {
  /**
   * Gets the current user
   */
  currentUser: GetUserAvatar_currentUser;
}
