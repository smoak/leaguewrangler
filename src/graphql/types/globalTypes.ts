export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type Query = {
  __typename?: 'Query';
  /** Gets the current user */
  currentUser: User;
  /** Get a single event */
  event?: Maybe<Event>;
  /** Get events for a team */
  events: Array<Maybe<Event>>;
  /** Get teams for a user */
  teams: Array<Maybe<Team>>;
};


export type QueryEventArgs = {
  eventId: Scalars['Int'];
  teamId: Scalars['Int'];
};


export type QueryEventsArgs = {
  teamId: Scalars['Int'];
};


export type QueryTeamsArgs = {
  dashboardOnly?: InputMaybe<Scalars['Boolean']>;
};

/** A user */
export type User = {
  __typename?: 'User';
  displayName: Scalars['String'];
  emailAddress1: Scalars['String'];
  events: Array<Maybe<Event>>;
  firstName: Scalars['String'];
  fullName: Scalars['String'];
  lastName: Scalars['String'];
  profilePhotoFullUrl: Scalars['String'];
  profilePhotoSmallUrl: Scalars['String'];
  profilePhotoThumbnailUrl: Scalars['String'];
  userId: Scalars['Int'];
};


/** A user */
export type UserEventsArgs = {
  dashboardOnly?: InputMaybe<Scalars['Boolean']>;
};

/** An event */
export type Event = {
  __typename?: 'Event';
  endTimestamp?: Maybe<Scalars['Int']>;
  eventId: Scalars['Int'];
  eventType: Scalars['String'];
  location?: Maybe<Location>;
  seasonId: Scalars['Int'];
  seasonName?: Maybe<Scalars['String']>;
  startTimestamp?: Maybe<Scalars['Int']>;
  status: Scalars['String'];
  team: Team;
  title: Scalars['String'];
  viewerRsvpStatus: RsvpStatus;
};

/** A location */
export type Location = {
  __typename?: 'Location';
  address?: Maybe<Scalars['String']>;
  googleMapsDirectionsUrl?: Maybe<Scalars['String']>;
  googleMapsUrl?: Maybe<Scalars['String']>;
  isPublic: Scalars['Boolean'];
  locationId: Scalars['Int'];
  name: Scalars['String'];
};

/** A team */
export type Team = {
  __typename?: 'Team';
  name: Scalars['String'];
  photoFullUrl?: Maybe<Scalars['String']>;
  photoSmallUrl?: Maybe<Scalars['String']>;
  photoThumbnailUrl?: Maybe<Scalars['String']>;
  shortName: Scalars['String'];
  teamId: Scalars['Int'];
};

/** The RSVP status of a user */
export enum RsvpStatus {
  /** Indicates the user is available to attend */
  AVAILABLE = 'AVAILABLE',
  /** Indicates the user might attend */
  MAYBE = 'MAYBE',
  /** Indicates the user will not attend */
  NO = 'NO',
  /** Indicates the user has not RSVPed */
  NONE = 'NONE',
  /** Indicates the user will attend */
  YES = 'YES'
}

export type Mutation = {
  __typename?: 'Mutation';
  /** Create a user token */
  createUserToken?: Maybe<UserToken>;
  /** RSVP to an event */
  saveRsvp: Event;
};


export type MutationCreateUserTokenArgs = {
  password: Scalars['String'];
  username: Scalars['String'];
};


export type MutationSaveRsvpArgs = {
  comments?: InputMaybe<Scalars['String']>;
  eventId: Scalars['Int'];
  rsvpStatus: RsvpStatus;
  teamId: Scalars['Int'];
};

/** A user token */
export type UserToken = {
  __typename?: 'UserToken';
  /** A unique token for the user matched. Tokens are 36-character, lower-cased GUIDs. */
  token: Scalars['String'];
};
