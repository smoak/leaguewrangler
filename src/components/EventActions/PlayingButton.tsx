import { IconButton } from '@material-ui/core';
import PlayingIcon from '@material-ui/icons/CheckCircleOutline';
import RsvpedPlayingIcon from '@material-ui/icons/CheckCircleTwoTone';
import { FC, useCallback } from 'react';

import { RsvpStatus } from '../../graphql/types/globalTypes';

interface PlayingButtonProps {
  readonly isPlaying: boolean;
  readonly className: string;
  onRsvp(rsvpStatus: RsvpStatus): void;
}

const PlayingButton: FC<PlayingButtonProps> = ({ isPlaying, className, onRsvp }) => {
  const rsvpCallback = useCallback(() => {
    onRsvp(RsvpStatus.YES);
  }, [onRsvp]);
  const icon = isPlaying ? (
    <RsvpedPlayingIcon className={className} color="primary" />
  ) : (
    <PlayingIcon className={className} />
  );

  return (
    <IconButton aria-label="Playing" onClick={rsvpCallback}>
      {icon}
    </IconButton>
  );
};

export default PlayingButton;
