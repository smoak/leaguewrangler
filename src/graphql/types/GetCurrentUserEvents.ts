/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { RsvpStatus } from "./globalTypes";

// ====================================================
// GraphQL query operation: GetCurrentUserEvents
// ====================================================

export interface GetCurrentUserEvents_currentUser_events_team {
  __typename: "Team";
  shortName: string;
  photoThumbnailUrl: string | null;
}

export interface GetCurrentUserEvents_currentUser_events_location {
  __typename: "Location";
  name: string;
  address: string | null;
  googleMapsUrl: string | null;
}

export interface GetCurrentUserEvents_currentUser_events {
  __typename: "Event";
  eventId: number;
  title: string;
  endTimestamp: number | null;
  startTimestamp: number | null;
  viewerRsvpStatus: RsvpStatus;
  team: GetCurrentUserEvents_currentUser_events_team;
  location: GetCurrentUserEvents_currentUser_events_location | null;
}

export interface GetCurrentUserEvents_currentUser {
  __typename: "User";
  events: (GetCurrentUserEvents_currentUser_events | null)[];
}

export interface GetCurrentUserEvents {
  /**
   * Gets the current user
   */
  currentUser: GetCurrentUserEvents_currentUser;
}
