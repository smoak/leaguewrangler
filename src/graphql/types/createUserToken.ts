import * as Types from './globalTypes';

export type CreateUserTokenMutationVariables = Types.Exact<{
  username: Types.Scalars['String'];
  password: Types.Scalars['String'];
}>;


export type CreateUserTokenMutation = { __typename?: 'Mutation', createUserToken?: { __typename?: 'UserToken', token: string } | null };
