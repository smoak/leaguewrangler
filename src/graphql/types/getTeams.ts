import * as Types from './globalTypes';

export type GetTeamsQueryVariables = Types.Exact<{ [key: string]: never; }>;


export type GetTeamsQuery = { __typename?: 'Query', teams: Array<{ __typename?: 'Team', name: string, id: number } | null> };
