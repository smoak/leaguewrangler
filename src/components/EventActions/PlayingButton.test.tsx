import { IconButton } from '@material-ui/core';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import { RsvpStatus } from '../../graphql/types/globalTypes';
import PlayingButton from './PlayingButton';

describe('PlayingButton', () => {
  let component: ShallowWrapper;

  describe('when playing', () => {
    beforeEach(() => {
      component = shallow(<PlayingButton isPlaying={true} className="test" onRsvp={jest.fn()} />);
    });

    it('renders as expected', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when not playing', () => {
    beforeEach(() => {
      component = shallow(<PlayingButton isPlaying={false} className="test" onRsvp={jest.fn()} />);
    });

    it('renders as expected', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when the user rsvps', () => {
    let onRsvp: jest.Mock;

    beforeEach(() => {
      onRsvp = jest.fn();
      component = shallow(<PlayingButton isPlaying={true} className="test" onRsvp={onRsvp} />);

      component.find(IconButton).simulate('click');
    });

    it('calls `onRsvp` with YES', () => {
      expect(onRsvp).toHaveBeenCalledWith(RsvpStatus.YES);
    });
  });
});
