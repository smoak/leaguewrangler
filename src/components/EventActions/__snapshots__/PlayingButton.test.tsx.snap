// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayingButton when not playing renders as expected 1`] = `
<WithStyles(ForwardRef(IconButton))
  aria-label="Playing"
  onClick={[Function]}
>
  <Memo(ForwardRef(CheckCircleOutlineIcon))
    className="test"
  />
</WithStyles(ForwardRef(IconButton))>
`;

exports[`PlayingButton when playing renders as expected 1`] = `
<WithStyles(ForwardRef(IconButton))
  aria-label="Playing"
  onClick={[Function]}
>
  <Memo(ForwardRef(CheckCircleTwoToneIcon))
    className="test"
    color="primary"
  />
</WithStyles(ForwardRef(IconButton))>
`;
